cpp/src/parquet/encoding.cc

```diff
@@ -3382,6 +3385,20 @@ class DeltaByteArrayDecoderImpl : public DecoderImpl, virtual public TypedDecode
       if (ARROW_PREDICT_FALSE(static_cast<size_t>(prefix_len_ptr[i]) > prefix.length())) {
         throw ParquetException("prefix length too large in DELTA_BYTE_ARRAY");
       }
+      if (prefix_len_ptr[i] == 0) {
+        prefix = std::string_view{buffer[i]};
+        continue;
+      }
+      if (buffer[i].len == 0 && prefix.data() != last_value_.data()) {
+        // postfix length == 0, point to the prefix.
+        // If prefix.data() == last_value_.data(), buffer might point to
+        // the last_value_, and last_value_ will be changed at the end
+        // of GetInternal. So we need to copy the prefix to the buffer.
```

Comment:
This part is weird, I want to know what do you think of this... Currently I use a hack `prefix.data() != last_value_.data()` here.